aten/src/ATen/native/hammerblade/Resize.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <TH/THTensor.hpp>
+#include <ATen/hammerblade/HammerBladeContext.h>
+#include <ATen/native/hammerblade/Offload.h>
+
+namespace at { namespace native {
+
+static inline void HBStorage_resize(THStorage* storage, ptrdiff_t size) {
```

Comment:
We need to think about this more actually ... THStorage is legacy storage code for CPU. This implementation definitely works and I understand you implemented it this way since CPU/CUDA they do so, but I was wondering if we should stay away from TH* in our HammerBlade code.

I think `THTensor*` is the same as `TensorImpl*` and `THStorage*` is just `StorageImpl*`.

Reply:
Another option is we keep it like this in the PR, and I'll refactor it after merging

Reply:
Nice suggestion! True, I didn't realize the transition to legacy objects is unnecessary here!